samples/doc-snippets/src/main/java/com/example/api/StaticResourcesEndpoint.java

```diff
@@ -0,0 +1,31 @@
+package com.example.api;
+
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.HttpResponse;
+import akka.javasdk.annotations.Acl;
+import akka.javasdk.annotations.http.Get;
+import akka.javasdk.annotations.http.HttpEndpoint;
+import akka.javasdk.http.HttpResponses;
+
+
+@HttpEndpoint
+@Acl(allow = @Acl.Matcher(principal = Acl.Principal.ALL))
+public class StaticResourcesEndpoint {
+
+  // tag::static-resources-from-classpath[]
+
+  // provide a landing page from root
+  @Get("/index.html")
+  public HttpResponse index() {
+    return HttpResponses.resourceFromClassPath("/static/index.html");
+  }
+
+  // map in all the available packaged static resources under /static
+  // see src/main/resources in project for actual files
+  @Get("/static/**")
+  public HttpResponse webPageResources(HttpRequest request) {
+    return HttpResponses.resourceFromClassPath(request.getUri().getPathString());
```

Comment:
Could we add an overload that takes `HttpRequest` and does exactly this, as a convenient default?

Reply:
Sure, when considering the API I just expected that not wanting to tie the directory structure it comes from on the classpath to the endpoint path it is published at would not be most common. It's just what I did in this sample to keep it minimal.

Reply:
I think that can be a good convention anyway. In the discussion they suggested that just placing content in a predefined resources directory would expose it automatically, but I tried to convince that it at least should be via an endpoint.

Reply:
Now updated to use a predefined resource directory `src/main/resources/static-resources` which the `HttpResponses.staticResource()` method makes available.

Added an overload accepting the http request as well, but with the additional prefix to strip as a required parameter to allow users to map whatever HTTP path prefix to be the content root of the `static-resources` directory.